chore(routes): Update and create new routes

neorusse opened the pull request:
#### What does this PR do?
- Update and create new routes

#### Description of Task to be completed?
- Create group controller
- Update message controller
- Update user controller
- Create authentication for routes
- Update table schema
- Test route with postman

#### How should this be manually tested?
- Clone the repo
- Cd into the directory
- Run npm install
- Use postman to test routes

#### What are the relevant pivotal tracker stories?
- PT Story link - [#164682690](https://www.pivotaltracker.com/story/show/164682690)

Commit:
chore(routes): Update and create new routes

- Create group controller
- Update message controller
- Update user controller
- Create authentication for routes
- Update table schema

[Finishes #164682690]

Commit:
chore(Controller): Refactor message controller

- Save message in sent and inbox table
- Retrieve all received message

[Finishes #164682690]

tomiadebanjo commented on server/controllers/GroupController.js:
```diff
@@ -0,0 +1,125 @@
+   * @returns {object} object
+   */
+  async addUser(req, res) {
+    if (!req.body.group_name || !req.body.role) {
```
Nice work. You need to check the db to know if the `group` exists before running the insert query

Commit:
chore(Controllers): Add routes to controllers

- Update db schema
- Update routes
- Refactor controllers

[Finishes #164682690]
